lib/Compressor.js

```diff
@@ -159,10 +182,20 @@ export class Compressor extends Transformer {
   }
 
   _transformTypedValue({entries, termType, value, termInfo}) {
-    const EncoderClass = TYPE_ENCODERS.get(termType);
-    const encoder = EncoderClass && EncoderClass.createEncoder(
-      {value, transformer: this, termInfo});
-    if(encoder) {
+    let encoder;
+
+    if(this.typedLiteralTable.has(termType)) {
+      encoder = TypedLiteralEncoder.createEncoder({
+        value,
+        transformer: this,
+        termType
+      });
+    } else {
+      const EncoderClass = TYPE_ENCODERS.get(termType);
+      encoder = EncoderClass && EncoderClass.createEncoder(
+        {value, transformer: this, termInfo});
+    }
+    if(encoder !== undefined && encoder !== false) {
```

Comment:
I think this simplifies to:

```suggestion
    if(encoder) {
```